elastic/kibana
# [Fleet] Fix data view overwriting on logs-* and metrics-*

## Summary

Fixes https://github.com/elastic/kibana/issues/120340

Currently, when we install an integration through Fleet, we _always_ recreate the `logs-*` and `metrics-*` data view objects. This has the negative consequence of blowing away any script fields or runtime fields a user may have configured on these data views. 

Also, these index patterns are not compatible with cross-cluster search today. 

This PR addresses these issues by doing the following: 
- Separate the creation of data views from the generic asset import flow and pass `overwrite: false` when we import data views
- Update the `title` property of Fleet's managed data views to support cross-cluster search by default, e.g. `logs-*,*:logs-*` and `metrics-*,*:metrics-*`
- Add logic to "migrate" existing data views to this new pattern and use a vanity name (`Logs` and `Metrics`) as the CCR pattern is less human-readable

## How to test

1. Create a local environment off of `main`
2. Install some integrations, enroll some agents, ingest some data
3. Switch to this branch
4. Install a new integration 
5. Observe logs that confirm the logs-* and metrics-* data views are migrated
6. Verify the new `Logs` and `Metrics` labeled data views continue to work and show your data in Discover

## Screenshots

Before:
![image](https://github.com/elastic/kibana/assets/6766512/22c8e92c-7355-4161-b0c2-4c103b96b439)

After: 
![image](https://github.com/elastic/kibana/assets/6766512/3afdfc3b-2e7a-41e7-85ff-77c65b77fb54)

![image](https://github.com/elastic/kibana/assets/6766512/cce852fb-c24d-4028-821a-370a35b76668)

Migration logging:
![image](https://github.com/elastic/kibana/assets/6766512/88d543d5-fe88-4087-a20d-ec9f08999fc2)
